utilities/aap.py

```
@@ -0,0 +1,628 @@
+def teardown_awx() -> None:
+    """Remove AWX deployment via Helm and delete namespace.
+
+    Attempts both Helm uninstall and namespace deletion. Logs
+    failures but does not raise — teardown errors should not
+    mark passing tests as failures.
+    """
+    LOGGER.info("Tearing down AWX deployment")
+    try:
+        _run_shell(f"helm uninstall {AWX_HELM_RELEASE_NAME} --namespace {AWX_NAMESPACE}")
+    except subprocess.CalledProcessError as e:
+        LOGGER.warning(f"Helm uninstall failed (stderr: {e.stderr}): {e}")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)